t/unit/app/test_app.py

```diff
@@ -385,6 +386,37 @@ def test_pending_configuration__raises_ImproperlyConfigured(self):
         with self.Celery() as app:
             assert not self.app.conf.task_always_eager
 
+    def test_pending_configuration__ssl_settings(self):
+        with self.Celery(broker='foo://bar',
+                         broker_use_ssl={
+                             'ssl_cert_reqs': ssl.CERT_REQUIRED,
+                             'ssl_ca_certs': '/path/to/ca.crt',
+                             'ssl_certfile': '/path/to/client.crt',
+                             'ssl_keyfile': '/path/to/client.key'}, 
+                         redis_backend_use_ssl = {
+                             'ssl_cert_reqs': ssl.CERT_REQUIRED,
```

Comment:
Should we also check what happens if someone uses the string form?

Reply:
Yes, I guess an extra check for that would be sensible. in `unit/app/test_app.py` it's simply checking that values added via the `broker_use_ssl` and `redis_backend_use_ssl` parameters when creating an instance of the `Celery` object end up in the config so I think here the content of the parameters is not significant. However, if ssl_cert_reqs is provided in string form, it should end up in the config in constant form after processing so I've added a test `test_backend_ssl_certreq_str` to the redis backend tests in `t/unit/backends/test_redis.py`